packages/next/src/client/script.tsx

```diff
@@ -4,7 +4,7 @@ import ReactDOM from 'react-dom'
 import React, { useEffect, useContext, useRef } from 'react'
 import { ScriptHTMLAttributes } from 'react'
 import { HeadManagerContext } from '../shared/lib/head-manager-context'
-import { DOMAttributeNames } from './head-manager'
+import { setBasicAttributesFromProps } from './head-manager'
```

Comment:
Can you move this function into a separate file? The way it's set up now is that it will end up pulling in `head-manager` in App Router too.

After that the PR looks good to land to me.